graphql/lessonplan/mutations.ts

```diff
@@ -0,0 +1,219 @@
+import { Resolver, Mutation, Arg, Authorized, Ctx, InputType, Field, Int, ObjectType, registerEnumType } from 'type-graphql';
+import { MaxLength, ArrayMaxSize, validateOrReject } from 'class-validator';
+import { Role } from '../authorizations';
+import { GraphQLContext } from '../context';
+import { getLogger } from '../../common/logger/logger';
+import { generateLessonPlan } from '../../common/lessonplan/generate';
+import { course_subject_enum, pupil_state_enum, school_schooltype_enum } from '@prisma/client';
+import { ApolloError, UserInputError } from 'apollo-server-express';
+
+const logger = getLogger(`LessonPlan Mutations`);
+
+// Maximum prompt length (in characters)
+const MAX_PROMPT_LENGTH = 10000;
+
+// Maximum number of files
+const MAX_FILES = 10;
+
+// Register the course_subject_enum as a GraphQL enum
+registerEnumType(course_subject_enum, {
+    name: 'CourseSubjectEnum',
+    description: 'The subject of the course',
+});
+
+// Register the school_schooltype_enum as a GraphQL enum
+registerEnumType(school_schooltype_enum, {
+    name: 'SchoolSchoolTypeEnum',
+    description: 'The type of school',
+});
+
+// Create a new enum for expected output fields
+enum LessonPlanField {
+    TITLE = 'title',
+    LEARNING_GOAL = 'learningGoal',
+    AGENDA_EXERCISES = 'agendaExercises',
+    ASSESSMENT = 'assessment',
+    HOMEWORK = 'homework',
+    RESOURCES = 'resources',
+}
+
+// Register the LessonPlanField enum
+registerEnumType(LessonPlanField, {
+    name: 'LessonPlanField',
+    description: 'Fields that can be generated for a lesson plan',
+});
+
+@InputType()
+class GenerateLessonPlanInput {
+    @Field(() => [String], { nullable: true })
+    @ArrayMaxSize(MAX_FILES, { message: `Number of files must not exceed ${MAX_FILES}` })
+    fileUuids?: string[];
+
+    @Field(() => course_subject_enum)
+    subject: course_subject_enum;
+
+    @Field(() => Int)
+    grade: number;
+
+    @Field(() => Int)
+    duration: number;
+
+    @Field(() => pupil_state_enum)
+    state: pupil_state_enum;
+
+    @Field(() => String)
+    @MaxLength(MAX_PROMPT_LENGTH, { message: `Prompt must not exceed ${MAX_PROMPT_LENGTH} characters` })
+    prompt: string;
+
+    @Field(() => [LessonPlanField], { nullable: true })
+    expectedOutputs?: LessonPlanField[];
```

Comment:
Hm, here we repeat the selected output again as parameters like `generateLessonPlan(expectedOutputs: ["some", "field"], ...) { some, field }`, correct? If yes, maybe we can add the `@Fields()` decorator proposed in  https://github.com/MichalLytek/type-graphql/issues/10#issuecomment-573061281 and then just derive the fields to produce from the query ...

Reply:
But I slightly wonder whether it is worth the effort, dont we always request all fields in the frontend?